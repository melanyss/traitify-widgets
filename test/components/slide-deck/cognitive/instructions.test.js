import Component from "components/slide-deck/cognitive/instructions";
import Practice from "components/slide-deck/cognitive/practice";
import ComponentHandler, {act} from "support/component-handler";
import {useResizeMock} from "support/mocks";

jest.mock("components/slide-deck/cognitive/practice", () => (() => <div className="mock">Practice</div>));

describe("Instructions", () => {
  let props;

  beforeEach(() => {
    props = {
      onStart: jest.fn().mockName("onStart"),
      translate: jest.fn().mockName("translate").mockImplementation((value) => value)
    };
  });

  describe("start", () => {
    let component;

    beforeEach(() => {
      component = new ComponentHandler(<Component {...props} />);
      component.act(() => component.findByText("cognitive_instructions_step_1_button").props.onClick());
      component.act(() => component.findByText("cognitive_instructions_step_2_button").props.onClick());
      component.act(() => component.instance.findByType(Practice).props.onFinish());
    });

    it("calls onStart", () => {
      component.act(() => component.findByText("cognitive_instructions_step_4_button").props.onClick());

      expect(props.onStart).toHaveBeenCalledWith({disability: false});
    });

    it("passes disability", () => {
      component.act(() => component.instance.findByProps({id: "traitify-disability"}).props.onChange());
      component.act(() => component.findByText("cognitive_instructions_step_4_button").props.onClick());

      expect(props.onStart).toHaveBeenCalledWith({disability: true});
    });
  });

  describe("videos", () => {
    let component;

    useResizeMock();

    describe("horizontal", () => {
      beforeEach(() => {
        act(() => window.resizeTo(1200, 800));
        component = new ComponentHandler(<Component {...props} />);
      });

      it("renders component", () => {
        expect(component.tree).toMatchSnapshot();
      });

      it("updates to vertical", () => {
        component.act(() => window.resizeTo(600, 800));

        expect(component.tree).toMatchSnapshot();
      });
    });

    describe("vertical", () => {
      beforeEach(() => {
        act(() => window.resizeTo(600, 800));
        component = new ComponentHandler(<Component {...props} />);
      });

      it("renders component", () => {
        expect(component.tree).toMatchSnapshot();
      });

      it("updates to horizontal", () => {
        component.act(() => window.resizeTo(1200, 800));

        expect(component.tree).toMatchSnapshot();
      });
    });
  });

  it("renders step 1", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders step 2", () => {
    const component = new ComponentHandler(<Component {...props} />);
    component.act(() => component.findByText("cognitive_instructions_step_1_button").props.onClick());

    expect(component.tree).toMatchSnapshot();
  });

  it("renders step 3", () => {
    const component = new ComponentHandler(<Component {...props} />);
    component.act(() => component.findByText("cognitive_instructions_step_1_button").props.onClick());
    component.act(() => component.findByText("cognitive_instructions_step_2_button").props.onClick());

    expect(component.tree).toMatchSnapshot();
  });

  it("renders step 4", () => {
    const component = new ComponentHandler(<Component {...props} />);
    component.act(() => component.findByText("cognitive_instructions_step_1_button").props.onClick());
    component.act(() => component.findByText("cognitive_instructions_step_2_button").props.onClick());
    component.act(() => component.instance.findByType(Practice).props.onFinish());

    expect(component.tree).toMatchSnapshot();
  });
});
