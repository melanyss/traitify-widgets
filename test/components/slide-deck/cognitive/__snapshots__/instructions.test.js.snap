// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions renders step 1 1`] = `
<div
  className="instructions"
>
  <h1>
    cognitive_instructions_step_1_heading
  </h1>
  <p>
    cognitive_instructions_step_1_text
  </p>
  <video
    autoPlay={true}
    disableremoteplayback="true"
    loop={true}
    muted={true}
    playsInline={true}
  >
    <source
      src="https://cdn.traitify.com/images/cognitive/instructions-h.mp4"
      type="video/mp4"
    />
  </video>
  <button
    className="btnBlue"
    onClick={[Function]}
    type="button"
  >
    cognitive_instructions_step_1_button
  </button>
</div>
`;

exports[`Instructions renders step 2 1`] = `
<div
  className="instructions"
>
  <h2>
    cognitive_instructions_step_2_heading
  </h2>
  <p>
    cognitive_instructions_step_2_text
  </p>
  <video
    autoPlay={true}
    disableremoteplayback="true"
    loop={true}
    muted={true}
    playsInline={true}
  >
    <source
      src="https://cdn.traitify.com/images/cognitive/practice-example-h.mp4"
      type="video/mp4"
    />
  </video>
  <button
    className="btnBlue"
    onClick={[Function]}
    type="button"
  >
    cognitive_instructions_step_2_button
  </button>
</div>
`;

exports[`Instructions renders step 3 1`] = `
<div
  className="mock"
>
  Practice
</div>
`;

exports[`Instructions renders step 4 1`] = `
<div
  className="instructions"
>
  <h2>
    cognitive_instructions_step_4_heading
  </h2>
  <p>
    cognitive_instructions_step_4_text
  </p>
  <label
    htmlFor="traitify-disability"
  >
    cognitive_instructions_disability_text
    <input
      checked={false}
      id="traitify-disability"
      name="disability"
      onChange={[Function]}
      type="checkbox"
    />
  </label>
  <button
    className="btnBlue"
    onClick={[Function]}
    type="button"
  >
    cognitive_instructions_step_4_button
  </button>
</div>
`;

exports[`Instructions videos horizontal renders component 1`] = `
<div
  className="instructions"
>
  <h1>
    cognitive_instructions_step_1_heading
  </h1>
  <p>
    cognitive_instructions_step_1_text
  </p>
  <video
    autoPlay={true}
    disableremoteplayback="true"
    loop={true}
    muted={true}
    playsInline={true}
  >
    <source
      src="https://cdn.traitify.com/images/cognitive/instructions-h.mp4"
      type="video/mp4"
    />
  </video>
  <button
    className="btnBlue"
    onClick={[Function]}
    type="button"
  >
    cognitive_instructions_step_1_button
  </button>
</div>
`;

exports[`Instructions videos horizontal updates to vertical 1`] = `
<div
  className="instructions"
>
  <h1>
    cognitive_instructions_step_1_heading
  </h1>
  <p>
    cognitive_instructions_step_1_text
  </p>
  <video
    autoPlay={true}
    disableremoteplayback="true"
    loop={true}
    muted={true}
    playsInline={true}
  >
    <source
      src="https://cdn.traitify.com/images/cognitive/instructions-v.mp4"
      type="video/mp4"
    />
  </video>
  <button
    className="btnBlue"
    onClick={[Function]}
    type="button"
  >
    cognitive_instructions_step_1_button
  </button>
</div>
`;

exports[`Instructions videos vertical renders component 1`] = `
<div
  className="instructions"
>
  <h1>
    cognitive_instructions_step_1_heading
  </h1>
  <p>
    cognitive_instructions_step_1_text
  </p>
  <video
    autoPlay={true}
    disableremoteplayback="true"
    loop={true}
    muted={true}
    playsInline={true}
  >
    <source
      src="https://cdn.traitify.com/images/cognitive/instructions-v.mp4"
      type="video/mp4"
    />
  </video>
  <button
    className="btnBlue"
    onClick={[Function]}
    type="button"
  >
    cognitive_instructions_step_1_button
  </button>
</div>
`;

exports[`Instructions videos vertical updates to horizontal 1`] = `
<div
  className="instructions"
>
  <h1>
    cognitive_instructions_step_1_heading
  </h1>
  <p>
    cognitive_instructions_step_1_text
  </p>
  <video
    autoPlay={true}
    disableremoteplayback="true"
    loop={true}
    muted={true}
    playsInline={true}
  >
    <source
      src="https://cdn.traitify.com/images/cognitive/instructions-h.mp4"
      type="video/mp4"
    />
  </video>
  <button
    className="btnBlue"
    onClick={[Function]}
    type="button"
  >
    cognitive_instructions_step_1_button
  </button>
</div>
`;
